lib/modules/datasource/docker/index.ts

```diff
@@ -630,16 +630,17 @@ export class DockerDatasource extends Datasource {
     let url: string | null =
       `${registryHost}/${dockerRepository}/tags/list?n=${limit}`;
     url = ensurePathPrefix(url, '/v2');
-    const headers = await getAuthHeaders(
+    const headers = (await getAuthHeaders(
       this.http,
       registryHost,
       dockerRepository,
       url,
-    );
-    if (!headers) {
-      logger.debug('Failed to get authHeaders for getTags lookup');
-      return null;
+    )) ?? { 'Content-Type': 'application/json' }; // if {} headers are provided, error will not be caught
+
+    if (headers['Content-Type'] === 'application/json') {
+      logger.debug('No authentication headers found, using default headers');
     }
```

Comment:
This assumes that no other host will ever return this header - is that a correct assumption?

Reply:
I don't think so

Reply:
Hi 

I made a similar change [here](https://github.com/hhomar/renovate/commit/e09a197463cbe8dc95dc65294277df0c6b8178bc), but didn't have time to submit it, and we've moved away from the ECR proxy. It throws a specific error for ECR max results, so it shouldn't affect other hosts.

Regardless of the best approach to take, I think the [ECR test](https://github.com/hhomar/renovate/commit/e09a197463cbe8dc95dc65294277df0c6b8178bc#diff-fd8051887597eeeda06bb09f0d0ec65ba11534ddcd0b9cfe80bb85abd3c16f01) should be updated as part of this PR as well. The test currently mocks a 200 response for the auth header call, instead of a 405, which is why this regression wasn't caught.